main/src/main/scala/sbt/MainLoop.scala

```diff
@@ -222,9 +224,18 @@ object MainLoop {
         val (restoreTerminal, termState) = channelName.flatMap(exchange.channelForName) match {
           case Some(c) =>
             val prevTerminal = ITerminal.set(c.terminal)
+            val needsJLine3 =
+              if (Project.isProjectLoaded(state)) Project.extract(state).get(Keys.needsJLine3)
+              else Nil
+            val closeJLine = if (needsJLine3.contains(exec.commandLine)) {
```

Comment:
Does this work with slash syntax or other composable commands like `~`, `;`, and `+`?

Reply:
No. I didn’t think of that.